source/helpers/transport-json.js

```diff
@@ -23,30 +23,56 @@ export default class TransportJSON extends Transport {
       this.emit('logged', log);
     });
     const sanitizedLogMessage = stripAnsi(log.message.replace('\n', ' ').trim());
-    if (log.level === 'info' || log.level === 'warn') {
-      if (log.isMessage) this.state.message += sanitizedLogMessage;
-      else this.state.logs.push(sanitizedLogMessage);
-    } else if (log.level === 'debug') {
-      this.state.debugModeLogs.push(`${log.callerFunction}: ${sanitizedLogMessage} ${log.ms}`);
-    } else if (log.level === 'error' && log.errorCode) {
-      if (!this.state.status !== 'error') {
+    switch (true) {
+      case log.level === 'info' || log.level === 'warn':
+        if (log.isMessage) this.state.message += sanitizedLogMessage;
```

Comment:
Please enforce to use curly braces with even one-line `if`s. Because if you have to add more statements, and forgot to add the curly braces back, then you'll be doomed